src/openrct2-ui/windows/Player.cpp

```diff
@@ -265,21 +266,23 @@ void window_player_overview_mouse_down(rct_window* w, rct_widgetindex widgetInde
 
 void window_player_overview_dropdown(rct_window* w, rct_widgetindex widgetIndex, int32_t dropdownIndex)
 {
-    int32_t player = network_get_player_index(static_cast<uint8_t>(w->number));
-    if (player == -1)
+    const auto playerId = static_cast<uint8_t>(w->number);
+    const auto playerIdx = network_get_player_index(playerId);
+    if (playerIdx == -1)
     {
         return;
     }
     if (dropdownIndex == -1)
     {
         return;
     }
-    int32_t group = network_get_group_id(dropdownIndex);
-    auto playerSetGroupAction = PlayerSetGroupAction(w->number, group);
+    const auto groupId = network_get_group_id(dropdownIndex);
+    const auto windowHandle = std::make_pair(w->classification, w->number);
+    auto playerSetGroupAction = PlayerSetGroupAction(playerId, groupId);
     playerSetGroupAction.SetCallback([=](const GameAction* ga, const GameActions::Result* result) {
```

Comment:
Maybe we should change the capture so that we explicitly only get `windowHandle`
```suggestion
    playerSetGroupAction.SetCallback([windowHandle](const GameAction* ga, const GameActions::Result* result) {
```

That would make it harder for us to sneak the usage of the pointer again.

Reply:
True, ill change it.